### PR TITLE
added rds dbcluster changes for handlers

### PR DESCRIPTION
**Pull request by mayankta**
*Issue #, if available:*

*Description of changes:*
* added rds dbcluster changes for handlers
* updated schema
* updated the create,update,delete handlers

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java [hunk 1]
```diff
@@ -6,14 +6,11 @@
...
+import software.amazon.awssdk.services.rds.model.DbClusterNotFoundException;
 import software.amazon.cloudformation.exceptions.CfnNotFoundException;
 import software.amazon.cloudformation.exceptions.CfnNotStabilizedException;
 import software.amazon.cloudformation.exceptions.TerminalException;
-import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
-import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
-import software.amazon.cloudformation.proxy.ProxyClient;
-import software.amazon.cloudformation.proxy.ProgressEvent;
-import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.*;
```
Comment (ammokhov) [inline aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java:13]: let's not use wildcard

File: aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/DeleteHandler.java [hunk 2]
```diff
@@ -1,26 +1,146 @@
...
+                        if (StringUtils.isNullOrEmpty(snapshotIdentifier)) {
+                            snapshotIdentifier = IdentifierUtils.generateResourceIdentifier(
+                                    Optional.ofNullable(request.getStackId()).orElse(STACK_NAME),
+                                    SNAPSHOT_PREFIX + Optional.ofNullable(request.getLogicalResourceIdentifier()).orElse(RESOURCE_IDENTIFIER),
+                                    request.getClientRequestToken(),
+                                    SNAPSHOT_MAX_LENGTH
+                            );
+                        }
+
+                        final String finalSnapshotIdentifier = snapshotIdentifier;
```
Comment (ammokhov) [inline aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/DeleteHandler.java:62]: why do you need another variable, when you already have `snapshotIdentifier`

**Review by ammokhov — APPROVED** [review]

File: aws-rds-dbcluster/aws-rds-dbcluster.json [hunk 3]
```diff
@@ -43,6 +43,11 @@
       "description": "The name of your database. If you don't provide a name, then Amazon RDS won't create a database in this DB cluster. For naming constraints, see Naming Constraints in the Amazon RDS User Guide.",
       "type": "string"
     },
+    "GlobalClusterIdentifier": {
+      "description": "If you are configuring an Aurora global database cluster and want your Aurora DB cluster to be a secondary member in the global database cluster, specify the global cluster ID of the global database cluster. To define the primary database cluster of the global cluster, use the AWS::RDS::GlobalCluster resource.\n\nIf you aren't configuring a global database cluster, don't specify this property.",
+      "type": "string",
+      "pattern": "^[a-zA-Z]{1}(?:-?[a-zA-Z0-9]){0,62}$"
+    },
```
Comment (moataz-mhmd) [inline aws-rds-dbcluster/aws-rds-dbcluster.json:50]: I see `CopyTagsToSnapshot` is missing. Which may break customer stacks that use this feature. Could you add it?

ref: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html

File: aws-rds-dbcluster/docs/readendpoint.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+}
+</pre>
+
+### YAML
+
+<pre>
+</pre>
+
+## Properties
+
```
Comment (moataz-mhmd) [inline aws-rds-dbcluster/docs/readendpoint.md:20]: nit: Can you run `pre-commit run --all-files`

**Comment by osdrv** [comment]
Closed in favor of https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/pull/86

**Pull request closed without merging**